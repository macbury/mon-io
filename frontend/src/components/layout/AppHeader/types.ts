import { HeaderProps } from 'react-navigation-stack'

interface IAppHeaderPropBase {
  topLevel?: boolean
  children?: any
  title?: string
  hideTitle?: boolean
  tabs?: boolean
  forceShowHeader?: boolean
}

export type AppHeaderProp = IAppHeaderPropBase & HeaderProps