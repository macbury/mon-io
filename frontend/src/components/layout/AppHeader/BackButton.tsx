import React from 'react'
import { useTheme } from 'styled-components/native'
import { Appbar } from 'react-native-paper'

interface IBackButtonProps {
  onPress()
}

export default function BackButton({ onPress } : IBackButtonProps) {
  const {
    colors: {
      icon
    }
  } : any = useTheme()

  return (
    <Appbar.BackAction
      color={icon}
      onPress={onPress} />
  )
}