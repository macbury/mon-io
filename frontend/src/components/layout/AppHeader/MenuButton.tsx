import React from 'react'
import { useTheme } from 'styled-components/native'
import { Appbar } from 'react-native-paper'

interface IMenuButtonProps {
  onPress()
}

export default function MenuButton({ onPress } : IMenuButtonProps) {
  const {
    colors: {
      icon
    }
  } : any = useTheme()

  return (
    <Appbar.Action
      icon="menu"
      color={icon}
      onPress={onPress} />
  )
}