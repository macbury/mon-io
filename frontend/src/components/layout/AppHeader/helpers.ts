import { AppHeaderProp } from './types'

export function getTitleFromNavigation(props : AppHeaderProp) : string {
  let title = props.title
  if (!title) {
    title = props.scene?.descriptor?.options?.title
  }
  return title
}