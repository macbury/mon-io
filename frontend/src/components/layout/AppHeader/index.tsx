import React from 'react'
import { StatusBar  } from 'react-native'
import { Appbar } from 'react-native-paper'
import styled, { useTheme } from 'styled-components/native'
import { useTranslation } from 'react-i18next'
import NetworkProgressBar from '../../UnstyledProgressBar/NetworkProgressBar'
import { AppHeaderProp } from './types'
import { getTitleFromNavigation } from './helpers'
import BackButton from './BackButton'
import MenuButton from './MenuButton'

const ProgressBar = styled(NetworkProgressBar)`
  position: absolute;
  left: 0;
  right: 0;
  bottom: -1px;
  z-index: 2;
`

const AppHeaderContainer = styled(Appbar.Header)`
  margin-top: ${StatusBar.currentHeight}px;
  background: ${(props) => props.theme.headerBackground};
  border-bottom-color: ${({ theme }) => theme.headerBorderColor};
  border-bottom-width: 1px;
`

export default function AppHeader(props : AppHeaderProp) {
  const theme = useTheme()
  const { hideTitle, tabs } = props
  const { t } = useTranslation()
  const goBack = () => {
    props.navigation.pop()
  }

  const appHeaderStyle = tabs ? { elevation: 0, borderBottomWidth: 0 } : { elevation: 0, borderBottomWidth: 1 }
  const canGoBack = props.navigation.state.routes.length > 1
  const title = getTitleFromNavigation(props)

  return (
    <AppHeaderContainer dark style={appHeaderStyle}>
      {canGoBack ? <BackButton onPress={goBack} /> : <MenuButton onPress={(props.navigation as any).toggleDrawer} />}
      {!hideTitle && <Appbar.Content title={t(title)} color={theme?.colors?.text} />}
      {props.children}
      <ProgressBar />
    </AppHeaderContainer>
  )
}