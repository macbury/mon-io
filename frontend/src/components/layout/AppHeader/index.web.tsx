import React, { useCallback } from 'react'
import { Appbar } from 'react-native-paper'
import { useTranslation } from 'react-i18next'
import styled, { useTheme } from 'styled-components/native'

import Mobile from '../../responsive/Mobile'
import Desktop from '../../responsive/Desktop'
import NetworkProgressBar from '../../UnstyledProgressBar/NetworkProgressBar'
import { AppHeaderProp } from './types'
import { getTitleFromNavigation } from './helpers'
import BackButton from './BackButton'
import MenuButton from './MenuButton'

const AppHeaderContainer = styled(Appbar.Header)`
  background: ${({ theme }) => theme.headerBackground};
  border-bottom-width: 1px;
  border-bottom-color: ${({ theme }) => theme.headerBorderColor};
`

const ProgressBar = styled(NetworkProgressBar)`
  position: absolute;
  left: 0;
  right: 0;
  bottom: -1px;
  z-index: 2;
`

export default function AppHeader(props : AppHeaderProp) {
  const theme = useTheme()
  const { navigation, hideTitle, tabs } = props
  const { t } = useTranslation()

  const canGoBack = navigation.state?.routes?.length > 1
  const title = getTitleFromNavigation(props)
  const appHeaderStyle = tabs ? { elevation: 0, borderBottomWidth: 0 } : { elevation: 0, borderBottomWidth: 1 }
  const goBack = useCallback(() => {
    navigation.pop()
  }, [navigation, canGoBack])

  const { forceShowHeader } = props.scene.descriptor.options as any

  return (
    <React.Fragment>
      <Mobile>
        <AppHeaderContainer dark style={appHeaderStyle}>
          {canGoBack ? <BackButton onPress={goBack} /> : <MenuButton onPress={(navigation as any).toggleDrawer} />}
          {!hideTitle && <Appbar.Content title={t(title)} color={theme.colors.text} />}
          {props.children}
          <ProgressBar />
        </AppHeaderContainer>
      </Mobile>
      <Desktop>
        {(!props.topLevel && canGoBack || forceShowHeader) && (<AppHeaderContainer style={appHeaderStyle} dark>
          {!forceShowHeader && canGoBack && <BackButton onPress={goBack} />}
          {!hideTitle && <Appbar.Content color={theme.colors.text} title={t(title)} />}
          {props.children}
          <ProgressBar />
        </AppHeaderContainer>)}
      </Desktop>
    </React.Fragment>
  )
}