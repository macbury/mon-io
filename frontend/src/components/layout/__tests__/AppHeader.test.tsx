import React from 'react'
import { shallow } from 'enzyme'
import snapshot from 'enzyme-to-json'

import AppHeader from '../AppHeader'

it('renders correctly', () => {
  const props = {
    theme: {
      colors: {
        text: '#000'
      }
    },
    navigation: {
      state: {
        routes: []
      }
    },
    scene: {
      descriptor: {
        options: {
          headerTitle: 'New header title'
        }
      }
    }
  }

  const component = shallow(<AppHeader {...props}/>)
  expect(snapshot(component)).toMatchSnapshot()
})