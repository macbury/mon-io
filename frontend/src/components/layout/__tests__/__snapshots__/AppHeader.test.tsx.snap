// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(withTheme(Appbar.Header))
  dark={true}
  style={
    Object {
      "borderBottomWidth": 1,
      "elevation": 0,
    }
  }
>
  <MenuButton />
  <withTheme(Appbar.Content)
    title=""
  />
  <Styled(NetworkProgressBar) />
</Styled(withTheme(Appbar.Header))>
`;
